### PR TITLE
🌱 Revert to plain kcp

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR undoes most of #1256, to revert to using plain kcp v0.11.0 instead of the patched variant that disables the Token controller for ServiceAccounts.

## Related issue(s)

This is part of the campaign #1274

### CODE REVIEW

File: pkg/cliplugins/kubestellar/syncer-gen/edgesync.go [hunk 1]
```diff
@@ -459,14 +468,14 @@ func (o *EdgeSyncOptions) enableSyncerForWorkspace(ctx context.Context, config *
...
-	}, metav1.CreateOptions{}); err != nil && !apierrors.IsAlreadyExists(err) {
+	}, metav1.CreateOptions{FieldManager: fieldManager}); err != nil && !apierrors.IsAlreadyExists(err) {
 		return "", "", nil, err
 	}
 
 	// Wait for the service account to be updated with the name of the token secret
 	tokenSecretName := ""
 	var serviceAccount *corev1.ServiceAccount
-	_ = wait.PollImmediateWithContext(ctx, 100*time.Millisecond, 3*time.Second, func(ctx context.Context) (bool, error) {
+	_ = wait.PollImmediateWithContext(ctx, 300*time.Millisecond, o.ServiceAccountTokenWait, func(ctx context.Context) (bool, error) {
```
Comment (waltforme) [inline pkg/cliplugins/kubestellar/syncer-gen/edgesync.go:478]: This 0.3 second duration seems to be different from line 485 ("we wait for 3 seconds").

File: pkg/cliplugins/kubestellar/syncer-gen/edgesync.go [hunk 2]
```diff
@@ -500,24 +500,62 @@ func (o *EdgeSyncOptions) enableSyncerForWorkspace(ctx context.Context, config *
...
+		tokenSecret, err = kubeClient.CoreV1().Secrets(namespace).Create(ctx, tokenSecret, metav1.CreateOptions{FieldManager: fieldManager})
 		if err != nil {
 			return "", "", nil, fmt.Errorf("failed to create ServiceAccount token secret: %w", err)
 		}
-	} else {
+		tokenSecretName = tokenSecret.Name
+	}
+	var fetchError error
+	var saTokenBytes []byte
+	err = wait.PollImmediateWithContext(ctx, 300*time.Millisecond, o.ServiceAccountTokenWait, func(ctx context.Context) (bool, error) {
```
Comment (waltforme) [inline pkg/cliplugins/kubestellar/syncer-gen/edgesync.go:512]: 0.3 seconds here also.

File: docs/content/Getting-Started/user-quickstart-test.md [hunk 3]
```diff
@@ -384,6 +399,11 @@ how to create, but not overwrite/update a synchronized resource
 <br>
 
 ## Tear it all down
+``` {.bash .hide-me}
+: not really
```
Comment (waltforme) [inline docs/content/Getting-Started/user-quickstart-test.md:403]: I'm not following this line here. May I have some more explanations?

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-start-kcp.md [hunk 4]
```diff
@@ -6,40 +6,23 @@ ways: as bare processes on whatever host you are using to run this
...
-                false ;;
-  esac
-  kcp_version=v0.11.0
-  trap "rm kcp.tar.gz kcp-plugins.tar.gz" EXIT
-  curl -SL -o kcp.tar.gz "https://github.com/kubestellar/kubestellar/releases/download/v0.12.0/kcp_0.11.0_${os_type}_${arch_type}.tar.gz"
-  curl -SL -o kcp-plugins.tar.gz "https://github.com/kcp-dev/kcp/releases/download/${kcp_version}/kubectl-kcp-plugin_${kcp_version//v}_${os_type}_${arch_type}.tar.gz"
-  tar -xzf kcp-plugins.tar.gz
-  tar -xzf kcp.tar.gz
-)
+make build
```
Comment (waltforme) [inline docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-start-kcp.md:25]: When I was testing example 1 locally I found that downloading kcp released binaries to be much faster than build from kcp source. Should we do downloading here?

File: pkg/cliplugins/kubestellar/syncer-gen/edgesync.go (hunk 1 above)
Comment (waltforme) [reply pkg/cliplugins/kubestellar/syncer-gen/edgesync.go:478]: Oh never mind, I see that the previous 3s timeout is now configurable by using `ServiceAccountTokenWait`.

File: docs/content/Getting-Started/user-quickstart-test.md (hunk 3 above)
Comment (MikeSpreitzer) [reply docs/content/Getting-Started/user-quickstart-test.md:403]: For a contributor doing testing, it is sometimes counter-productive to have the test destroy the state it developed. I find it helpful to be able to do further investigations after the test completed.

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-start-kcp.md (hunk 4 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-start-kcp.md:25]: Perhaps. I am not happy with the big bunch of bash used uniquely here to do that downloading. Remember that part of the virtue of example1 is that it tests the copy of Kubestellar that it is embedded in, not a different copy fetched from GitHub or elsewhere.

File: docs/content/Getting-Started/user-quickstart-test.md (hunk 3 above)
Comment (MikeSpreitzer) [reply docs/content/Getting-Started/user-quickstart-test.md]: I have revised this PR to accomplish the goal in a non-hacky way.

Comment (waltforme) [reply docs/content/Getting-Started/user-quickstart-test.md]: I see. Thank you!

**Pull request merged**